Loop/Managers/LoopAppManager.swift

```diff
@@ -297,9 +305,47 @@ extension LoopAppManager: AlertPresenter {
         rootViewController?.topmostViewController.present(viewControllerToPresent, animated: animated, completion: completion)
     }
 
-    func dismiss(animated: Bool, completion: (() -> Void)?) {
+    func dismissTopMost(animated: Bool, completion: (() -> Void)?) {
         rootViewController?.topmostViewController.dismiss(animated: animated, completion: completion)
     }
+
+    func dismissAlert(_ alertToDismiss: UIAlertController, animated: Bool, completion: (() -> Void)?) {
+        if rootViewController?.topmostViewController == alertToDismiss {
+            dismissTopMost(animated: animated, completion: completion)
+        } else {
+            // check if the alert to dismiss is presenting another alert (and so on)
+            // calling dismiss() on an alert presenting another alert will only dismiss the presented alert
+            // (and any other alerts presented by the presented alert)
+
+            // get the stack of presented alerts that would be undesirably dismissed
+            var presentedAlerts: [UIAlertController] = []
+            var currentAlert = alertToDismiss
+            while let presentedAlert = currentAlert.presentedViewController as? UIAlertController {
+                presentedAlerts.append(presentedAlert)
+                currentAlert = presentedAlert
+            }
+
+            if presentedAlerts.isEmpty {
+                alertToDismiss.dismiss(animated: animated, completion: completion)
+            } else {
+                // Do not animate any of these view transitions, since the alert to dismiss is not at the top of the stack
+                // dismiss all the child presented alerts
+                alertToDismiss.dismiss(animated: false) {
+                    // dismiss the desired alert
+                    alertToDismiss.dismiss(animated: false) {
+                        // present the child alerts that were undesirably dismissed
+                        for alert in presentedAlerts {
+                            if alert == presentedAlerts.last {
+                                self.present(alert, animated: false, completion: completion)
+                            } else {
+                                self.present(alert, animated: false, completion: nil)
```

Comment:
Shouldn't you wait to present the next one until after this one completes presentation? That is, present first and in its completion block present second and in its completion block present third... present last and call the passed in completion?

Reply:
Good point.  You are right that this is needed to maintain the order. In my use case there was only 2 alerts presented, so the order was always maintained. I'll update.